src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -31210,4 +31362,1076 @@ static Expression<Func<Func<string, string>>> Test()
+            //         protected void M() { }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "M").WithArguments("E.extension(int).M()").WithLocation(5, 24),
+            // (6,23): error CS9302: 'E.extension(int).P': new protected member declared in an extension block
+            //         protected int P { get => 0; set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "P").WithArguments("E.extension(int).P").WithLocation(6, 23),
+            // (7,35): error CS9302: 'E.extension(int).P2.get': new protected member declared in an extension block
+            //         public int P2 { protected get => 0; set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "get").WithArguments("E.extension(int).P2.get").WithLocation(7, 35),
+            // (8,45): error CS9302: 'E.extension(int).P3.set': new protected member declared in an extension block
+            //         public int P3 { get => 0; protected set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "set").WithArguments("E.extension(int).P3.set").WithLocation(8, 45));
+    }
+
+    [Fact]
+    public void Validation_Modifiers_ProtectedInternal()
+    {
+        string source = """
+static class E
+{
+    extension(int i)
+    {
+        protected internal void M() { }
+        protected internal int P { get => 0; set { } }
+        public int P2 { protected internal get => 0; set { } }
+        public int P3 { get => 0; protected internal set { } }
+    }
+}
+""";
+        var comp = CreateCompilation(source);
+        comp.VerifyEmitDiagnostics(
+            // (5,33): error CS9302: 'E.extension(int).M()': new protected member declared in an extension block
+            //         protected internal void M() { }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "M").WithArguments("E.extension(int).M()").WithLocation(5, 33),
+            // (6,32): error CS9302: 'E.extension(int).P': new protected member declared in an extension block
+            //         protected internal int P { get => 0; set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "P").WithArguments("E.extension(int).P").WithLocation(6, 32),
+            // (7,44): error CS9302: 'E.extension(int).P2.get': new protected member declared in an extension block
+            //         public int P2 { protected internal get => 0; set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "get").WithArguments("E.extension(int).P2.get").WithLocation(7, 44),
+            // (8,54): error CS9302: 'E.extension(int).P3.set': new protected member declared in an extension block
+            //         public int P3 { get => 0; protected internal set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "set").WithArguments("E.extension(int).P3.set").WithLocation(8, 54));
+    }
+
+    [Fact]
+    public void Validation_EntryPoint_01()
```

Comment:
>Validation_EntryPoint_01 [](http://example.com/codeflow?start=16&length=24)

Is this a dupe of ```Validation_EntryPoint_02```?

Reply:
The C# snippet is the same so I could consolidate, but `Validation_EntryPoint_02` uses `WithMainTypeName` which exercises a different path in `FindEntryPoint`.

Reply:
I missed the ```WithMainTypeName``` difference